tests/sentry/workflow_engine/processors/test_workflow.py

```diff
@@ -343,6 +343,58 @@ def test_enqueues_workflow_any_logic_type(self):
         )
         assert project_ids[0][0] == self.project.id
 
+    def test_skips_enqueuing_any(self):
+        # skips slow conditions if the condition group evaluates to True without evaluating them
+        assert self.workflow.when_condition_group
+        self.workflow.when_condition_group.update(
+            logic_type=DataConditionGroup.Type.ANY_SHORT_CIRCUIT
+        )
+
+        self.create_data_condition(
+            condition_group=self.workflow.when_condition_group,
+            type=Condition.EVENT_FREQUENCY_COUNT,
+            comparison={
+                "interval": "1h",
+                "value": 100,
+            },
+            condition_result=True,
+        )
+
+        triggered_workflows = evaluate_workflow_triggers({self.workflow}, self.job)
+        assert triggered_workflows == {self.workflow}
+        project_ids = buffer.backend.get_sorted_set(
+            WORKFLOW_ENGINE_BUFFER_LIST_KEY, 0, self.buffer_timestamp
+        )
+        assert len(project_ids) == 0
+
+    def test_skips_enqueuing_all(self):
```

Comment:
if there is at least 1 fast condition that is not met and the logic type is `all`, we don't have to evaluate slow conditions either